### PR TITLE
Community docs

### PR DESCRIPTION
**Pull request by richardcase**
Adding documents to the repo that peope generally expect from
a OSS project. This is still WIP.

- [x] Code of conduct
- [x] Contributing
- [x] Update README
- [x] Issue template
- [x] Pull request template
- [x] MAINTAINERS/OWNERS
- [x] humans.txt

### CODE REVIEW

**Comment by errordeveloper** [comment]
+@dholbach

**Comment by richardcase** [comment]
@errordeveloper @dholbach - i thought i'd start adding these. Its mostly based on other Weaveworks repos but feel free to suggest any changes.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# How to Contribute
+
+*eksctl* is [Mozilla Public License Version 2.0 licenced](LICENSE.MPL-2.0) and accepts contributions via GitHub
```
Comment (dholbach) [inline CONTRIBUTING.md]: This is not quite true, `eksctl` is under [Apache 2.0 license](https://github.com/weaveworks/eksctl/blob/master/LICENSE). Only some vendored files are under MPL-2.0.

File: CODE-OF-CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Community Code of Conduct
+
+Eksctl follows the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).
+
+Instances of abusive, harassing, or otherwise unacceptable behavior
+may be reported by contacting a Weaveworks project maintainer, or
+Alexis Richardson alexis@weave.works.
```
Comment (dholbach) [inline CODE-OF-CONDUCT.md]: "Alexis Richardson <alexis@weave.works>" maybe?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+pull requests. This document outlines some of the conventions on development
```
Comment (dholbach) [inline CONTRIBUTING.md]: "…on *the* development workflow…"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+The project uses Slack. Join the [Weave community](https://weaveworks.github.io/community-slack/) Slack workspace and use the [#eksctl](https://weave-community.slack.com/messages/eksctl/) channel.
+
+## Getting Started
+
+- Fork the repository on GitHub
+- Read the [README](README.md) for getting started as a user and learn how/where to ask for help 
+- If you want to contribute as a developer, continue reading this document for further instructions
+- Play with the project, submit bugs, submit pull requests!
+
```
Comment (dholbach) [inline CONTRIBUTING.md]: Encourage to ask questions on Slack if they get stuck. Maybe add a link to ["good first issues"](https://github.com/weaveworks/eksctl/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+## Contribution workflow
+
+This is a rough outline of how to prepare a contribution:
+
+- Create a topic branch from where you want to base your work (usually branched from master).
+- Make commits of logical units.
+- Make sure your commit messages are in the proper format (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- If you changed code:
+   - add automated tests to cover your changes
```
Comment (dholbach) [inline CONTRIBUTING.md]: Maybe add a link to a good test case to illustrate what's expected?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+These things will make a PR more likely to be accepted:
+
+ * a well-described requirement
+ * tests for new code
+ * tests for old code!
+ * new code and tests follow the conventions in old code and tests
+ * a good commit message (see below)
+
+In general, we will merge a PR once two maintainers have endorsed it.
```
Comment (dholbach) [inline CONTRIBUTING.md]: Is this true? Who are the two (or more?) designated maintainers right now?

File: CONTRIBUTING.md (hunk 1 above)
Comment (richardcase) [reply CONTRIBUTING.md]: I'll change that back to what it was...thanks.

File: CODE-OF-CONDUCT.md (hunk 2 above)
Comment (richardcase) [reply CODE-OF-CONDUCT.md]: Yeah i'd agree

File: CONTRIBUTING.md (hunk 4 above)
Comment (richardcase) [reply CONTRIBUTING.md]: Good idea, i'll add.

File: CONTRIBUTING.md (hunk 5 above)
Comment (richardcase) [reply CONTRIBUTING.md]: That would be useful, i'll add.

File: CONTRIBUTING.md (hunk 6 above)
Comment (richardcase) [reply CONTRIBUTING.md]: I'll change to `we will merge a PR once a maintainer has reviewed and approved it`

File: CODE-OF-CONDUCT.md [hunk 7]
```diff
@@ -3,5 +3,5 @@
 Eksctl follows the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).
 
 Instances of abusive, harassing, or otherwise unacceptable behavior
-may be reported by contacting a Weaveworks project maintainer, or
-Alexis Richardson alexis@weave.works.
\ No newline at end of file
+may be reported by contacting a *eksctk* project maintainer, or
```
Comment (dholbach) [inline CODE-OF-CONDUCT.md]: *eksctl*

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -65,7 +70,7 @@ These things will make a PR more likely to be accepted:
  * new code and tests follow the conventions in old code and tests
  * a good commit message (see below)
 
-In general, we will merge a PR once two maintainers have endorsed it.
+In general, we will merge a PR once a maintainer has reviwed and approved it.
```
Comment (dholbach) [inline CONTRIBUTING.md]: *reviewed*

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+We gratefully welcome improvements to documentation as well as to code.
+
+# Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. No action from you is required, but it's a good idea to see the
+[DCO](DCO) file for details before you start contributing code to eksctl.
```
Comment (dholbach) [inline CONTRIBUTING.md]: `DCO` file needs to imported still.

Comment (richardcase) [reply CONTRIBUTING.md]: DCO file has been added, thanks.

**Comment by richardcase** [comment]
@dholbach - i added the missing DCO file. Do you think we need to add anything else at this stage?

**Comment by errordeveloper** [comment]
@dholbach @richardcase do you think this is still WIP?

**Comment by richardcase** [comment]
@errordeveloper - its good to go from my part. I've changed the title.

**Comment by dholbach** [comment]
Excellent work on this @richardcase!

**Pull request merged**